pkg/vault/test/foundry/BufferVaultPrimitive.t.sol

```diff
@@ -139,6 +147,34 @@ contract BufferVaultPrimitiveTest is BaseVaultTest {
         );
     }
 
+    function testDepositMaliciousRouter() public {
+        // Deposit will not take the underlying tokens, keeping the approval, so the wrapper can use vault approval to
+        // drain the whole vault.
+        waDAI.setMaliciousWrapper(true);
+
+        uint256 vaultBalance = dai.balanceOf(address(vault));
+
+        vault.unlock(
+            abi.encodeWithSelector(
+                BufferVaultPrimitiveTest.erc4626MaliciousHook.selector,
+                BufferWrapOrUnwrapParams({
+                    kind: SwapKind.EXACT_IN,
+                    direction: WrappingDirection.WRAP,
+                    wrappedToken: IERC4626(address(waDAI)),
+                    amountGivenRaw: vaultBalance,
+                    limitRaw: 0,
+                    userData: bytes("")
+                })
+            )
+        );
+
+        vm.expectRevert(
+            abi.encodeWithSelector(IERC20Errors.ERC20InsufficientAllowance.selector, address(waDAI), 0, vaultBalance)
+        );
+        vm.prank(address(waDAI));
+        dai.transferFrom(address(vault), address(waDAI), vaultBalance);
```

Comment:
I'd (also) check that the allowance granted by the vault is 0 (maybe a good check to add in other wrapper tests as well?)

Reply:
ah nvm; it's encoded in the error.

As an alternative, maybe just check that the allowance is 0 instead of calling `transferFrom` and expecting revert?

Reply:
Done!